libafl/src/events/launcher.rs

```diff
@@ -716,7 +716,8 @@ where
                                 self.time_obs.clone(),
                             )?;
 
-                            self.main_run_client.take().unwrap()(state, c_mgr, *bind_to)
+                            self.main_run_client.take().unwrap()(state, c_mgr, *bind_to)?;
```

Comment:
Shouldn't the run loop already reutrn ShuttingDown?

Reply:
Nope

Reply:
Why not? Also, can you change `noed` -> `node` in the main_run_client fn while you're having a PR open? :D

Reply:
where is this? I can't seem to find it.

Reply:
It is not necessary that the run_client function propagates Error::ShuttingDown, it may decide to handle it manually and return Ok(())